### PR TITLE
feat(sentry): Group Sentry issues by scraper

### PR DESCRIPTION
**Pull request by grossir**
Uses Sentry's fingerprinting of events to force a grouping. 
For scraping and parsing errors, the desired grouping is by the data's source scraper, which can be identified by the juriscraper module. If the module is not available, by the court id

Solves #3399

### CODE REVIEW

File: cl/settings/third_party/sentry.py [hunk 1]
```diff
@@ -13,6 +15,31 @@
 ignore_logger("internetarchive.session")
 ignore_logger("internetarchive.item")
 
+
+def fingerprint_event(
+    event: Dict[str, Any], hint: Dict[str, Any]
```
Comment (flooie) [inline cl/settings/third_party/sentry.py:20]: typing can be lower case I think.  Atleast for dict.

File: cl/settings/third_party/sentry.py [hunk 2]
```diff
@@ -13,6 +15,31 @@
 ignore_logger("internetarchive.session")
 ignore_logger("internetarchive.item")
 
+
+def fingerprint_event(
```
Comment (flooie) [inline cl/settings/third_party/sentry.py:19]: I think the name could be slightly more descriptive...

maybe something like ... `fingerprint_sentry_error`

File: cl/settings/third_party/sentry.py [hunk 3]
```diff
@@ -13,6 +15,31 @@
...
+) -> Dict[str, Any]:
+    """If a fingerprint key is present in "extra", pass it to Sentry
+
+    This function expects error.log(extra={"fingerprint": []})
+    to be a List
+
+    The fingerprint will be a "group ID" on Sentry, forcing all events
+    with that key to be grouped on a single issue
+
+    :param event: event dict to be sent to Sentry
```
Comment (flooie) [inline cl/settings/third_party/sentry.py:30]: I guess im just mostly confused by this docstring- think you can take a crack at rewriting it.

**Review by flooie — CHANGES_REQUESTED** [review]
This is great.  @grossir 

Can you just rework the docstrings on the fingerprint function.  I found it hard to understand

**Comment by grossir** [comment]
Hi @flooie please review again. Sorry about the messy commit history; I rebased the branch so that it was up to date with main but now all the commits are included in the PR. Still the files changed remain the same

- cl/settings/third_party/sentry.py
- cl/scrapers/management/commands/cl_scrape_opinions.py
- cl/scrapers/utils.py
- cl/scrapers/tasks.py

**Comment by flooie** [comment]
> ## Suspect Issues
> This pull request was deployed and Sentry observed the following issues:
> 
> * ‼️ **SSLError: HTTPSConnectionPool(host='www.jud.ct.gov', port=443): Max retries exceeded with url: /external/su...** `cl.scrapers.management.commands.cl_scrape_opini...` [View Issue](https://freelawproject.sentry.io/issues/4908738955/?referrer=github-pr-bot)
> 
> Did you find this useful? React with a 👍 or 👎


this is in dev sentry - I was testing a known bug

**Pull request merged**